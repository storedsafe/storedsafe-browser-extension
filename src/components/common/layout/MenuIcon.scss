@import '../../../styles/variables.scss';

button:hover .menu-icon .menu-icon-bar,
button:focus .menu-icon .menu-icon-bar {
  background-color: $color-primary-3;
}

.menu-icon {
  $size: $menu-size - $spacing * 2;
  $bar-height: floor($size / 7);

  display: flex;
  width: $size;
  height: $size;
  flex-direction: column;
  justify-content: space-between;

  .menu-icon-bar {
    width: $size;
    height: $bar-height;
    background-color: $color-bg;
    transition: $transition-duration;
  }

  &.selected .menu-icon-bar-1 {
    transform: rotate(-45deg) translate(-$size * 0.3, $size * 0.3);
  }

  &.selected .menu-icon-bar-2 {
    opacity: 0;
  }

  &.selected .menu-icon-bar-3 {
    transform: rotate(45deg) translate(-$size * 0.3, -$size * 0.3);
  }
}
