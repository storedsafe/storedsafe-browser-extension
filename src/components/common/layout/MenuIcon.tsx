import React from 'react'
import './MenuIcon.scss'

interface MenuIconProps {
  selected: boolean
}

export const MenuIcon: React.FunctionComponent<MenuIconProps> = ({
  selected
}: MenuIconProps) => {
  return (
    <div className={`menu-icon${selected ? ' selected' : ''}`}>
      <div className='menu-icon-bar menu-icon-bar-1' />
      <div className='menu-icon-bar menu-icon-bar-2' />
      <div className='menu-icon-bar menu-icon-bar-3' />
    </div>
  )
}
