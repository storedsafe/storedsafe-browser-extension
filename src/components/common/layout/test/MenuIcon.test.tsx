import * as React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import { MenuIcon } from '../MenuIcon'
import pretty from 'pretty'

/**
 * Set up and tear down container to mount tested component in.
 * */
let container: Element = null
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div')
  document.body.appendChild(container)
})
afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

test('<MenuIcon selected=true />', () => {
  act(() => {
    render(<MenuIcon selected={true} />, container)
  })

  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test('<MenuIcon selected=false />', () => {
  act(() => {
    render(<MenuIcon selected={false} />, container)
  })

  expect(pretty(container.innerHTML)).toMatchSnapshot()
})
