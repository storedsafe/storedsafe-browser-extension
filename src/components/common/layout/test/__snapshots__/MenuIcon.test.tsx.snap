// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuIcon selected=false /> 1`] = `
"<div class=\\"menu-icon\\">
  <div class=\\"menu-icon-bar menu-icon-bar-1\\"></div>
  <div class=\\"menu-icon-bar menu-icon-bar-2\\"></div>
  <div class=\\"menu-icon-bar menu-icon-bar-3\\"></div>
</div>"
`;

exports[`<MenuIcon selected=true /> 1`] = `
"<div class=\\"menu-icon selected\\">
  <div class=\\"menu-icon-bar menu-icon-bar-1\\"></div>
  <div class=\\"menu-icon-bar menu-icon-bar-2\\"></div>
  <div class=\\"menu-icon-bar menu-icon-bar-3\\"></div>
</div>"
`;
